### PR TITLE
Java smoothing should now be fully interruptible

### LINKED ISSUE #471: Smooth file truncated by IGB, should be investigated
Check result in Integrated Genome Browser.
Smoothing should use a moving window and compute the mean or median of the window and assign to the rounded (any direction) center coordinate. This can be rewritten in Java with relative ease.

### PR DESCRIPTION
**Pull request by Mintey**
Closes #471.

### CODE REVIEW

File: src/smoothing/SmoothingAndStep.java [hunk 1]
```diff
@@ -80,6 +80,10 @@ public double smoothing(int[] params, String inPath, String outPath,
 
                 if((strLine = br.readLine()) != null){
                     while(!addLine(strLine)){
+                        if(Thread.currentThread().isInterrupted()){
```
Comment (23Skidoo) [inline src/smoothing/SmoothingAndStep.java]: Doing this check in the processing loop feels suboptimal. What was the problem with the external process solution again?

Comment (Mintey) [inline src/smoothing/SmoothingAndStep.java]: Indeed it feels suboptimal, and I'm trying to find a smoother way of doing it,  havent found anything yet though. From what I've read, it seems the only "solution" you can do is optimize the calls to `Thread.currentThread().isInterrupted()`. 

The problem I was having with the external process solution was either it didn't run the process at all, or it produced empty files.

However, this solves the problem, not elegantly, but it does solve it (as far as I can test, the rest is up to the researchers to check (validity of calculations and such)).

Comment (23Skidoo) [inline src/smoothing/SmoothingAndStep.java]: > The problem I was having with the external process solution was either it didn't run the process at all, or it produced empty files.

Why was it doing that? Should be more or less easy to fix.

Comment (Mintey) [inline src/smoothing/SmoothingAndStep.java]: I'll look into a better solution once I have the time (such as running it as an external process).

**Pull request merged**